### PR TITLE
Add proxy docs

### PR DESCRIPTION
**Pull request by eriknelson**

### CODE REVIEW

File: docs/proxy.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+[edit the broker's DeploymentConfig and set the HTTP_PROXY, HTTPS_PROXY, and NO_PROXY environment variables](https://docs.openshift.com/container-platform/3.7/install_config/http_proxies.html#setting-environment-variables-in-pods).
+It's common that APB pods themselves may require external access via proxy as well.
+If the broker recognizes it has a proxy configuration, it will transparently
+apply this env vars to the APB pods that it spawns. As long as the modules used
+within the APB respect proxy configuration via environment variable, the APB
+will also use these settings to perform its work. 
+
+Finally, it's possible the services spawned by the APB may also require external
+network access via proxy. The APB *must* be authored to set these environment variables
+explicitly if if recognizes them in its own execution environment, or the cluster
```
Comment (chengzhang1016) [inline docs/proxy.md]: Dup "if" in here: "f if recognizes them"

**Comment by chengzhang1016** [comment]
@eriknelson Others LGTM. Thx.

File: docs/proxy.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Running ASB behind a proxy
+
+It is recommended to review the [Openshift Documentation ](https://docs.openshift.com/container-platform/3.7/install_config/http_proxies.html)
+and to configure a cluster accordingly before attempting to run the broker behind
+a proxy. When running an Ansible Service Broker inside of a proxied Openshift cluster,
```
Comment (jmrodri) [inline docs/proxy.md]: Probably best to start using "Ansible Broker".

TYPO: Openshift should be OpenShift.

File: docs/proxy.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+it's important to understand its core concepts and consider them within the context
+of a proxy used for external network access.
+
+As an overview, the broker itself runs as a pod within the cluster. It has a requirement
+for external network access depending on how its registries have been configured.
+To configure the broker for external access via proxy, the cluster operator must
+[edit the broker's DeploymentConfig and set the HTTP_PROXY, HTTPS_PROXY, and NO_PROXY environment variables](https://docs.openshift.com/container-platform/3.7/install_config/http_proxies.html#setting-environment-variables-in-pods).
+It's common that APB pods themselves may require external access via proxy as well.
+If the broker recognizes it has a proxy configuration, it will transparently
+apply this env vars to the APB pods that it spawns. As long as the modules used
```
Comment (jmrodri) [inline docs/proxy.md]: TYPO: this -> these

**Pull request merged**